tools/graph_sanity_check.py

```diff
@@ -0,0 +1,288 @@
+    Returns
+    -------
+        dict of list : dict of names of the edge features of different edge types
+    """
+    fname_dict = {}
+    for feat_name in edge_feat_names:
+        feat_info = feat_name.split(":")
+        assert len(feat_info) == 2, (
+            f"Unknown format of the feature name: {feat_name}, " \
+            "must be EDGE_TYPE:FEAT_NAME"
+        )
+        etype = feat_info[0].split(",")
+        assert len(etype) == 3, (
+            f"EDGE_TYPE should have 3 strings: {etype}, " \
+            "must be NODE_TYPE,EDGE_TYPE,NODE_TYPE:"
+        )
+        etype = ":".join(etype)
+        assert etype not in fname_dict, (
+            f"You already specify the feature names of {etype} " \
+            f"as {fname_dict[etype]}"
+        )
+        # multiple features separated by ','
+        fname_dict[etype] = feat_info[1].split(",")
+    return fname_dict
+
+def gen_node_error_prefix(partition, ntype, fname):
+    """ Create an error prefix for reporting node feature/masks.
+
+        The Prefix will be in the following format nypte-fname-partition.
+        So when errors are reported we can easily sort it and the errors from the
+        same node type and same feature will be grouped.
+
+        Parameters
+        ----------
+        partition: int
+            Partition ID.
+        ntype: str
+            Node type.
+        fname: str
+            Feature name.
+
+        Return
+        ------
+            str: the error prefix
+    """
+    return f"[Node type: {ntype}][Feature Name: {fname}][Part {partition}]:"
+
+def gen_edge_error_prefix(partition, etype, fname):
+    """ Create an error prefix for reporting eode feature/masks.
+
+        The Prefix will be in the following format eypte-fname-partition.
+        So when errors are reported we can easily sort it and the errors from the
+        same eode type and same feature will be grouped.
+
+        Parameters
+        ----------
+        partition: int
+            Partition ID.
+        etype: tuple
+            Edge type.
+        fname: str
+            Feature name.
+
+        Return
+        ------
+            str: the error prefix
+    """
+    return f"[Edge type: {etype}][Feature Name: {fname}][Part {partition}]:"
+
+def verify_masks(mask, error_prefix):
+    """ Validate the data of a mask
+
+        Parameters
+        ----------
+        mask: tensor
+            Mask
+        error_prefix: str
+            Error report prefix
+
+        Return
+        ------
+            str: the error message if the mask is invalid otherwise None
+    """
+    try:
+        # DGL use nonzero_1d to translate a mask into indices of nodes/edges.
+        # so we check the mask using dgl.backend.nonzero_1d
+        mask = F.nonzero_1d(mask)
+    except:
+        return f"ERROR: {error_prefix} The mask is not loadable for loadable. " \
+                 "Ideally, the mask should be a 1D bool tensor."
+    return None
+
+def verify_feats(feat, error_prefix):
```

Comment:
In case when there are multiple errors in node/edge features, can we group these errors together and return them. Otherwise, users need to run this tool a few times to find out all of these errors.